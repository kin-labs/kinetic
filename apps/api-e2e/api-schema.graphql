# ------------------------------------------------------
# THIS FILE WAS AUTOMATICALLY GENERATED (DO NOT MODIFY)
# ------------------------------------------------------

type App {
  createdAt: DateTime!
  id: String!
  index: Int!
  name: String
  updatedAt: DateTime!
}

input AppCreateInput {
  index: Int!
  name: String!
}

input AppUpdateInput {
  name: String
}

type AuthToken {
  token: String!
}

"""
A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format.
"""
scalar DateTime

input LoginInput {
  email: String!
  password: String!
}

type Mutation {
  createApp(input: AppCreateInput!): App
  deleteApp(appId: String!): App
  login(input: LoginInput!): AuthToken
  logout: Boolean
  updateApp(appId: String!, input: AppUpdateInput!): App
}

type Query {
  app(appId: String!): App
  apps: [App!]
  me: User
  uptime: Float!
}

type User {
  avatarUrl: String!
  createdAt: DateTime!
  emails: [UserEmail!]
  id: String!
  name: String!
  role: UserRole
  updatedAt: DateTime!
  username: String!
}

type UserEmail {
  createdAt: DateTime!
  email: String!
  id: String!
  updatedAt: DateTime!
}

enum UserRole {
  Admin
  User
}
