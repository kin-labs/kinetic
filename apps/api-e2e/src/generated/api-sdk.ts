/* eslint-disable */
import gql from 'graphql-tag'
export type Maybe<T> = T | null
export type InputMaybe<T> = Maybe<T>
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] }
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> }
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> }
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string
  String: string
  Boolean: boolean
  Int: number
  Float: number
  /** A date-time string at UTC, such as 2019-12-03T09:54:33Z, compliant with the date-time format. */
  DateTime: any
}

export type AuthToken = {
  __typename?: 'AuthToken'
  token: Scalars['String']
}

export type LoginInput = {
  email: Scalars['String']
  password: Scalars['String']
}

export type Mutation = {
  __typename?: 'Mutation'
  login?: Maybe<AuthToken>
  logout?: Maybe<Scalars['Boolean']>
}

export type MutationLoginArgs = {
  input: LoginInput
}

export type Query = {
  __typename?: 'Query'
  me?: Maybe<User>
  uptime: Scalars['Float']
}

export type User = {
  __typename?: 'User'
  avatarUrl: Scalars['String']
  createdAt: Scalars['DateTime']
  emails?: Maybe<Array<UserEmail>>
  id: Scalars['String']
  name: Scalars['String']
  role?: Maybe<UserRole>
  updatedAt: Scalars['DateTime']
  username: Scalars['String']
}

export type UserEmail = {
  __typename?: 'UserEmail'
  createdAt: Scalars['DateTime']
  email: Scalars['String']
  id: Scalars['String']
  updatedAt: Scalars['DateTime']
}

export enum UserRole {
  Admin = 'Admin',
  User = 'User',
}

export const AuthTokenDetails = gql`
  fragment AuthTokenDetails on AuthToken {
    token
  }
`
export const UserDetails = gql`
  fragment UserDetails on User {
    id
    createdAt
    updatedAt
    avatarUrl
    name
    username
    role
  }
`
export const UserEmailDetails = gql`
  fragment UserEmailDetails on UserEmail {
    id
    createdAt
    updatedAt
    email
  }
`
export const Login = gql`
  mutation Login($input: LoginInput!) {
    login(input: $input) {
      ...AuthTokenDetails
    }
  }
  ${AuthTokenDetails}
`
export const Me = gql`
  query Me {
    me {
      ...UserDetails
    }
  }
  ${UserDetails}
`
export const Uptime = gql`
  query Uptime {
    uptime
  }
`
export type AuthTokenDetailsFragment = { __typename?: 'AuthToken'; token: string }

export type LoginMutationVariables = Exact<{
  input: LoginInput
}>

export type LoginMutation = { __typename?: 'Mutation'; login?: { __typename?: 'AuthToken'; token: string } | null }

export type MeQueryVariables = Exact<{ [key: string]: never }>

export type MeQuery = {
  __typename?: 'Query'
  me?: {
    __typename?: 'User'
    id: string
    createdAt: any
    updatedAt: any
    avatarUrl: string
    name: string
    username: string
    role?: UserRole | null
  } | null
}

export type UptimeQueryVariables = Exact<{ [key: string]: never }>

export type UptimeQuery = { __typename?: 'Query'; uptime: number }

export type UserDetailsFragment = {
  __typename?: 'User'
  id: string
  createdAt: any
  updatedAt: any
  avatarUrl: string
  name: string
  username: string
  role?: UserRole | null
}

export type UserEmailDetailsFragment = {
  __typename?: 'UserEmail'
  id: string
  createdAt: any
  updatedAt: any
  email: string
}

export interface PossibleTypesResultData {
  possibleTypes: {
    [key: string]: string[]
  }
}
const result: PossibleTypesResultData = {
  possibleTypes: {},
}
export default result
