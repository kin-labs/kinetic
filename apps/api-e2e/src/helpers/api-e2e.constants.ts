export const ADMIN_EMAIL = process.env.ADMIN_EMAIL
export const ADMIN_PASSWORD = process.env.ADMIN_PASSWORD
