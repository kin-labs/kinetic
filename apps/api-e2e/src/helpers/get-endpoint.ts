import { INestApplication } from '@nestjs/common'
import * as request from 'supertest'

export function getEndpoint(app: INestApplication, endpoint: string) {
  return request(app.getHttpServer()).get(endpoint)
}
