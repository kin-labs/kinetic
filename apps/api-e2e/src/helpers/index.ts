export * from './api-e2e.constants'
export * from './get-endpoint'
export * from './initialize-e2e-app'
export * from './run-query'
export * from './random-int'
