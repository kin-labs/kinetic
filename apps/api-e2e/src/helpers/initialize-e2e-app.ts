import { INestApplication, ValidationPipe } from '@nestjs/common'
import { Test, TestingModule } from '@nestjs/testing'
import { AppModule } from '@kin-kinetic/api-app-module'

export async function initializeE2eApp(): Promise<INestApplication> {
  const moduleFixture: TestingModule = await Test.createTestingModule({
    imports: [AppModule],
  }).compile()

  const app = moduleFixture.createNestApplication()
  app.useGlobalPipes(new ValidationPipe({ transform: true }))
  app.setGlobalPrefix('api')
  await app.init()
  return app
}
