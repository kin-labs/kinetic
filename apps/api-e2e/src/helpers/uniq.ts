import { getRandomInt } from './random-int'

export function uniq(prefix: string) {
  return `${prefix}${uniqInt()}`
}
export function uniqInt(): number {
  return Math.floor(Math.random() * 10000000)
}
export function randomAppIndex(): number {
  return getRandomInt(1_000, 1_000_000)
}
