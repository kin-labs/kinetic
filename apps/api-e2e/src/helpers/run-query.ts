import { INestApplication } from '@nestjs/common'
import { UserRole } from '@prisma/client'
import { ASTNode } from 'graphql'
import { print } from 'graphql/language/printer'
import * as request from 'supertest'
import { Login, LoginInput } from '../generated/api-sdk'
import { ADMIN_PASSWORD } from './api-e2e.constants'

export function runGraphQLQuery(
  app: INestApplication,
  query: ASTNode,
  variables?: Record<string, unknown>,
  { token }: { token?: string } = {},
) {
  return request(app.getHttpServer())
    .post('/graphql')
    .set('Authorization', token ? `Bearer ${token}` : null)
    .send({ query: print(query), variables: variables })
}

export function runGraphQLQueryAdmin(
  app: INestApplication,
  token: string,
  query: ASTNode,
  variables?: Record<string, unknown>,
) {
  return runGraphQLQuery(app, query, variables, { token })
}

export function runLoginQuery(app: INestApplication, email: string, role: UserRole) {
  const input: LoginInput = { password: ADMIN_PASSWORD, email }
  return runGraphQLQuery(app, Login, { input }).expect(200)
}
