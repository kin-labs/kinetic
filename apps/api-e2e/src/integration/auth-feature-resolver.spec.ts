import { INestApplication } from '@nestjs/common'
import { UserRole } from '@prisma/client'
import { Me } from '../generated/api-sdk'
import { runGraphQLQueryAdmin, runLoginQuery } from '../helpers'
import { initializeE2eApp } from '../helpers/'
import { ADMIN_EMAIL } from '../helpers/api-e2e.constants'

describe('Auth (e2e)', () => {
  let app: INestApplication
  let token: string | undefined

  beforeAll(async () => {
    app = await initializeE2eApp()
  })
  afterAll(async () => {
    return app.close()
  })

  describe('log in ', () => {
    it('should login with sample user: admin', async () => {
      const res = await runLoginQuery(app, ADMIN_EMAIL, UserRole.Admin)
      expect(res.body.data.login.token).toBeDefined()
      token = res.body.data.login.token
    })
  })

  describe('retrieve data of logged in user', () => {
    it('should retrieve data of user: admin', () => {
      return runGraphQLQueryAdmin(app, token, Me)
        .expect(200)
        .expect((res) => {
          expect(res).toHaveProperty('body.data')
          const data = res.body.data

          expect(data).toHaveProperty('me.id')

          // Remove any property that might change each run, snapshot it
          data.me.id = undefined
          data.me.createdAt = undefined
          data.me.updatedAt = undefined
          expect(data.me).toMatchSnapshot()

          // Pick properties specifically and test value.
          expect(data.me.role).toEqual(UserRole.Admin)
          expect(data.me.name).toEqual('Mogami Admin')
          expect(data.me.username).toEqual(ADMIN_EMAIL)
        })
    })
  })
})
