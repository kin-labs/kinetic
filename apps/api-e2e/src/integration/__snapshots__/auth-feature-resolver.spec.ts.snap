// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth (e2e) retrieve data of logged in user should retrieve data of user: admin 1`] = `
Object {
  "avatarUrl": "https://avatars.githubusercontent.com/u/82999948?v=4",
  "createdAt": undefined,
  "id": undefined,
  "name": "Mogami Admin",
  "role": "Admin",
  "updatedAt": undefined,
  "username": "mogami@kin.org",
}
`;
